### DISCUSSION TITLE
Property in store is not reactive

### DISCUSSION THREAD

**Discussion by webJose** [vuejs/pinia#2030] [Help and Questions] [answered]
### Reproduction

https://stackblitz.com/edit/github-nu748r?file=src%2FApp.vue

### Steps to reproduce the bug

1. Open the demo.
2. Look at the console.  A warning will read:

> watch source can only be a getter/effect function, a ref, a reactive object, or an array of these types.

The corresponding code is:

```typescript
const store = usePeopleStore();
// This watch doesn't work.  store.peopleCount is not reactive.
// It is my understanding that the state properties are reactive.
// Bug?
watch(store.peopleCount, (newVal) => {
  console.log('peopleCount changed: %d', newVal);
});
```

### Expected behavior

The call to `watch()` should happily accept the state property and the watch should run the callback function.

### Actual behavior

The `watch()` function complains, both  as Intellisense in VS Code as well as in runtime in the console log.  The callback function never runs.

### Additional information

So far, I have written this bug report from the "Pinia's store property is not reactive" standpoint, but it so happens that direct assignment of the store property seems to trigger the reactive nature.

So, bug in Pinia?  Or in Vue's `watch()`?  Or not a bug at all and I'm the idiot here?  Hehe.  Don't be afraid to tell me it's the last one.  I am just starting Vue.js, so I may very well be mistaken.

**Comment by posva** [comment, marked as answer]
It's

```ts
watch(() => store.peopleCount, (newVal) => {
  console.log('peopleCount changed: %d', newVal);
});
```

**Reply by webJose** [reply]
Hi @posva. I know I can do that.  The whole point is that I shouldn't need to do it like that.  The `watch()` function states that it works with reactive objects, and Pinia claims all store properties are reactive, yet, `watch()` says otherwise.

**Reply by posva** [reply]
`store.peopleCount` is not an object, it's likely a number, so it's as if you wrote `watch(20, ...)`, that's why it doesn't work

**Reply by webJose** [reply]
Yes, a number, you are correct.  Maybe then I misunderstand Pinia.

I can do:

```javascript
peopleStore.peopleCount = newValue;
```

And that seems to be reactive.  Plus, my mind tells me I read or saw in a tutorial that the store properties are reactive, and that reactivity is broken when we destructure.

You seem to be very sure about this, so I guess I won't take more of your time.  I'll maybe re-read Pinia's documentation and try to figure this one out by myself.

Yes, if I do:

```javascript
const { peopleCount } = storeToRefs(peopleStore);
watch(peopleCount, ....);
```

That works.  But this working only makes me think that the property is just a number like you say, but in my mind it contradicts the studies about Pinia, again, because somewhere I learned that the entire store is reactive.

**Reply by posva** [reply]
This should help: https://vuejs.org/guide/essentials/reactivity-fundamentals.html#limitations-of-reactive

**Comment by webJose** [comment]
Hey, I have a question, if you all don't mind, related to my initial topic.

I see the following in the Vue devtools:

![image](https://user-images.githubusercontent.com/25540842/222324937-10c95345-49f5-46b9-af01-9ec10b108581.png)

As pointed out by the red arrow, Vue devtools seems to be telling me that the `peopleCount` property is reactive.  Is it not???
